File: alg.js
class SSTF {
  constructor() {
    this.requests = [];
    this.pointer = 0; 
  }

  add(request) {
    this.requests.push(request);
  }

  get() {
    if(this.requests.length === 0) {
      return -1; 
    }

    let minDistance = Infinity;
    let closestIndex = -1;

    for(let i = 0; i < this.requests.length; i++) {
      if(i !== this.pointer) {
        let distance = Math.abs(this.requests[i] - this.requests[this.pointer]);
        if(distance < minDistance) {
          minDistance = distance;
          closestIndex = i;
        }
      }
    }

    this.pointer = closestIndex;
    return closestIndex;
  }
}


class SCAN {
  constructor() {
    this.requests = new Array(32).fill(0); 
    this.direction = 1; 
    this.pointer = 0;
  }

  init() {
    this.requests.fill(0);
    this.direction = 1; 
    this.pointer = 0; 
  }

  isEmpty() {
    return !this.requests.some(r => r === 1);
  }

  add(request) {
    this.requests[request] = 1;
  }

  get() {
    if(this.isEmpty()) {
      return -1;
    }

    while(true) {
      if(this.direction === 1) {
        for(; this.pointer < 32; this.pointer++) {
          if(this.requests[this.pointer] === 1) {
            this.requests[this.pointer] = 0;
            return this.pointer;
          }
        }
        this.pointer--;
        this.direction = 0;
      }
      else {
        for(; this.pointer >= 0; this.pointer--) {
          if(this.requests[this.pointer] === 1) {
            this.requests[this.pointer] = 0;
            return this.pointer;  
          }
        }
        this.pointer++;
        this.direction = 1;
      }
    }
  }
}

File: alg.c
#include <iostream>
#include <vector>
#include <stdlib.h>
using namespace std;

class SSTF {
    private:
        vector<int> sstf;
        int pointer = 0;
    public:
        void add(int input) {
            sstf.push_back(input);
            return;
        }
        int get(void) {
            int index = -1;
            int min_distance = 21470000;
            if(!sstf.empty()) {
                for(int i = 0; i < sstf.size(); i++) {
                    if(i != pointer) {
                        int distance = sstf.at(i) - sstf.at(pointer);
                        if(distance < min_distance) {
                            min_distance = distance;
                            index = i;
                            pointer = i;
                        }
                    }
                }
            }
            else {
                return -1;
            }
            return index;
        }
};

class scan {
    private:
        int array[32];
        int dir = 1;
        int pointer = 0;
    public:
        void init() {
            for(int i = 0; i < 32; i++) {
                array[i] = 0;
            }
            dir = 1;
            pointer = 0;
        }
        bool isEmpty() {
            for(int i = 0; i < 32; i++) {
                if(array[i] != 0) return true;
            }
            return false;
        }
        int add(int input) {
            array[input] = 1;
        }
        int get() {
            if(isEmpty) return -1;
            while(1) {
                if(dir == 1) {
                    for(; pointer < 32 ;pointer++) {
                        if(array[pointer] == 1) {
                            array[pointer] = 0;
                            return pointer;
                        }
                    }
                    pointer--; 
                    dir = 0;
                }
                if(dir == 0) {
                    for(; pointer >= 0 ; pointer--) {
                        if(array[pointer] == 1) {
                            array[pointer] = 0;
                            return pointer;
                        }
                    }
                    pointer++;
                    dir = 1;
                }
            }
            return -1;
        }

};



int main(void) {
    cout<<"mad madhacker\n";

    return 0;
}
